src/composables/websocket.ts

```diff
@@ -0,0 +1,172 @@
+   */
+  function subscribeToWebSocketChannel(channelName: WebSocketChannelTypes) {
+    if (
+      !channelName
+      || channels[channelName]
+      || !isWsConnected.value
+    ) {
+      return;
+    }
+
+    wsClient.send((JSON.stringify({
+      op: 'Subscribe',
+      payload: channelName,
+    } as IMiddlewareWebSocketSubscriptionMessage)));
+    channels[channelName] = true;
+  }
+
+  /**
+   * Handle websocket message event.
+   * @param {MessageEvent} message - The message event object.
+   */
+  function handleWebsocketMessage(message: MessageEvent) {
+    if (!message.data) {
+      return;
+    }
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers[data.subscription as WebSocketChannelTypes]
+      ) {
+        return;
+      }
+      // Call all subscribers for the channel
+      Object.values(subscribers[data.subscription as WebSocketChannelTypes]).forEach(
+        (subscriberCb: any) => subscriberCb(data.payload),
+      );
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  function subscribeToWebSocketChannels() {
+    // Add an event listener for messages
+    wsClient.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    (Object.keys(channels) as WebSocketChannelTypes[])
+      .forEach(subscribeToWebSocketChannel);
+  }
+
+  /**
+   * UnSubscribes websocket channels.
+   */
+  function unsubscribeFromWebSocketChannels() {
+    // Remove event listener for messages
+    wsClient.removeEventListener('message', handleWebsocketMessage);
+
+    // UnSubscribe all channels
+    Object.keys(channels).forEach((channelName: string) => {
+      channels[channelName as WebSocketChannelTypes] = false;
+    });
+  }
+
+  /**
+   * This function creates a new WebSocket client and attempts to connect to the specified server.
+   * @param {string} url - The URL of the WebSocket server to connect to
+   */
+  async function connectToWebSocketServer(url: string) {
+    if (wsClient) {
+      unsubscribeFromWebSocketChannels();
+
+      if (isWsConnected.value) {
+        wsClient.close();
+      }
+    }
+
+    isWsConnected.value = false;
+
+    wsClient = new WebSocket(url);
+
+    wsClient.onopen = () => {
+      isWsConnected.value = true;
+      subscribeToWebSocketChannels();
+    };
+
+    wsClient.onclose = () => {
+      isWsConnected.value = false;
+    };
+  }
+
+  /**
+   * Subscribes to a specific websocket channel
+   * @param {WebSocketChannelTypes} channel - The websocket channel to subscribe to.
+   * @param {Function} [callback=(payload: any) => payload]
+   *    - The function to be called when a message is received on the specified channel.
+   * @returns {Function}
+   *    - A function that can be called to unsubscribe the specified callback from the channel.
+   */
+  function watchChannelUpdates(
```

Comment:
What do you think about creating 3 more functions that would serve as a shortcuts to specific channels? For example:
```
const watchTransactionsUpdates = (payload) => watchChannelUpdates('Transactions', payload);
```
So whenever any of the components wants to watch a specific channel it would import this shortcut function instead of the `watchChannelUpdates` and the constant with the channel name.